libs/sdk-model/src/execution/executionDefinition/factory.ts

```diff
@@ -155,6 +157,49 @@ export function defWithSorting(definition: IExecutionDefinition, sorts: ISortIte
     return defSetSorts(definition, sorts);
 }
 
+/**
+ * Changes the postProcessing of a definition.
+ *
+ * This function MUST be used to implement IPreparedExecution.withPostProcessing();
+ *
+ * @param definition - execution definition to alter with postProcessing
+ * @param postProcessing - configuration that should be done with the data after they are obtained from the server
+ *  and before they are passed to the user
+ * @returns new execution with the specified postProcessing
+ * @public
+ */
+export function defWithPostProcessing(
+    definition: IExecutionDefinition,
+    postProcessing: IPostProcessing,
+): IExecutionDefinition {
+    return defSetPostProcessing(definition, postProcessing);
+}
+
+/**
+ * Changes the dateFormat of a postProcessing, other properties of postProcessing (if any) remain unchanged.
+ * This function will call defWithPostProcessing to update definition with the new postProcessing.
+ *
+ * @param definition - execution definition to alter with postProcessing
+ * @param dateFormat - Format to be applied to the dates in an AFM execution response.
+ * If dateFormat is empty, then the postProcessing will not be updated.
```

Comment:
I wonder if this is user-friendly. How would you remove dateFormat from an executionDefinition?